src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -204,13 +200,34 @@ public static Text of(Object... objects) {
                 } else {
                     // Unsupported TextAction
                 }
+
+                // Special content - does not use the specified Text formatting
+            } else if (obj instanceof TextRepresentable) {
+                builder.append(((TextRepresentable) obj).toText());
+            } else if (obj instanceof CatalogType) {
```

Comment:
I feel it would be better for the implementations to implement `TextRepresentable` as some are `Translatable`, not `TextRepresentable`.